src/models/threadpool.cpp

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "threadpool.h"
+
+namespace Generators {
+
+ThreadPool::ThreadPool(size_t num_threads) : num_threads_{num_threads} {}
+
+void ThreadPool::Compute(const std::function<void(size_t)>& func) {
+  for (size_t i = 0; i < num_threads_; ++i) {
+    threads_.emplace_back([&, i] { func(i); });
```

Comment:
I wonder if the overhead of thread creation/joining is significant relative to the actual computation?

Reply:
I am quite certain it is significant. But when I ran some tests on my windows arm64 machine, I saw a speedup (~0.8 tokens per second) relative to not running with the threads.
This implementation is quite rudimentary and will need to be enhanced. It would be great if we can leverage the ort threadpool from across the C API interface.